src/main/java/org/gridsuite/shortcircuit/server/service/ReportMapper.java

```diff
@@ -0,0 +1,151 @@
+        reporterModel.getReports().forEach(newReporter::report);
+        reporterModel.getSubReporters().forEach(reporter -> newReporter.addSubReporter(
+            switch (reporter.getTaskKey()) {
+                case "ShortCircuitAnalysis" -> forShortCircuitAnalysis(reporter);
+                default -> reporter;
+            }));
+        return newReporter;
+    }
+
+    /**
+     * Modify node with key {@code ShortCircuitAnalysis}
+     *
+     * @implNote we assume there will always be at least one modification
+     */
+    protected ReporterModel forShortCircuitAnalysis(@NonNull final ReporterModel reporterModel) {
+        final ReporterModel newReporter = new ReporterModel(reporterModel.getTaskKey(), reporterModel.getDefaultName(), reporterModel.getTaskValues());
+        reporterModel.getReports().forEach(newReporter::report);
+        reporterModel.getSubReporters().forEach(reporter -> newReporter.addSubReporter(
+            switch (reporter.getTaskKey()) {
+                case "generatorConversion" -> forGeneratorConversion(reporter);
+                case "courcirc" -> forCourcirc(reporter);
+                default -> reporter;
+            }));
+        return newReporter;
+    }
+
+    /**
+     * Modify node with key {@code generatorConversion}
+     */
+    protected ReporterModel forGeneratorConversion(@NonNull final ReporterModel reporterModel) {
+        return reporterModel;
+    }
+
+    /**
+     * Modify node with key {@code courcirc}
+     *
+     * @apiNote the main goal here is to concat specific log lines with a huge number of repetitions
+     * @implNote courcirc logs don't seem to use enumerate keys (REC_0,REC_1,...) neither values, so we need to test the message
+     */
+    protected ReporterModel forCourcirc(@NonNull final ReporterModel reporterModel) {
+        log.trace("courcirc logs detected, will analyse them...");
+        final ReporterModel newReporter = new ReporterModel(reporterModel.getTaskKey(), reporterModel.getDefaultName(), reporterModel.getTaskValues());
+        reporterModel.getSubReporters().forEach(newReporter::addSubReporter);
+        /* preparing */
+        final Pattern patternTransientReactanceTooLow = Pattern.compile("^(.+) +: transient reactance too low ==> generator ignored$", Pattern.CASE_INSENSITIVE);
+        final List<String> logsTransientReactanceTooLow = new ArrayList<>(newReporter.getReports().size());
+        /* analyze and compute logs in one pass */
+        for (final Report report : reporterModel.getReports()) { //we modify logs conditionally here
+            final Matcher matcherTransientReactanceTooLow = patternTransientReactanceTooLow.matcher(report.getDefaultMessage());
+            if (matcherTransientReactanceTooLow.matches()) { //we match line "X.ABCDEF1 : transient reactance too low ==> generator ignored"
+                logsTransientReactanceTooLow.add(matcherTransientReactanceTooLow.group(1));
```

Comment:
@SlimaneAmar do we want to keep the old reports, and just add one summary with a different verbosity ? Or remove the original reports entirely ?

Reply:
I think we should ask Pauline